exercises/baffling-birthdays/canonical-data.json

```
@@ -0,0 +1,176 @@
+      "property": "matchingBirthday",
+      "input": {
+        "birthdate1": "1997-08-04",
+        "birthdate2": "1963-11-23"
+      },
+      "expected": false
+    },
+    {
+      "uuid": "70b38cea-d234-4697-b146-7d130cd4ee12",
+      "description": "random birthdates return specified number of birthdays",
+      "scenarios": ["random"],
+      "property": "randomBirthdates",
+      "input": {},
+      "expected": "length == groupsize"
+    },
+    {
+      "uuid": "d9d5b7d3-5fea-4752-b9c1-3fcd176d1b03",
+      "description": "random birthdates have fixed year per batch",
+      "scenarios": ["random"],
+      "property": "randomBirthdates",
+      "input": {},
+      "expected": {
+        "years": {
+          "fixed": true
+        }
+      }
+    },
+    {
+      "uuid": "8da69a53-6900-4b63-897d-9f025f149fd2",
+      "description": "random birthdates have random year between batches",
+      "scenarios": ["random"],
+      "property": "randomBirthdates",
+      "input": {},
+      "expected": {
+        "years": {
+          "random": true
+        }
+      }
+    },
+    {
+      "uuid": "d1074327-f68c-4c8a-b0ff-e3730d0f0521",
+      "description": "random birthdates have random months",
+      "scenarios": ["random"],
+      "property": "randomBirthdates",
+      "input": {},
+      "expected": {
+        "months": {
+          "random": true
+        }
+      }
+    },
+    {
+      "uuid": "7df706b3-c3f5-471d-9563-23a4d0577940",
+      "description": "random birthdates have random days",
+      "scenarios": ["random"],
+      "property": "randomBirthdates",
+      "input": {},
+      "expected": {
+        "days": {
+          "random": true
+        }
+      }
+    },
+    {
+      "uuid": "ade37c87-f41d-4929-962a-286b4d1d048a",
+      "description": "has matching birthdays with matching birthday pair",
+      "property": "hasMatchingBirthdays",
+      "input": {
+        "birthdays": ["1970-01-19", "1975-06-03", "2003-01-19"]
+      },
+      "expected": true
+    },
+    {
+      "uuid": "1d155b33-c6e1-46b9-81fa-09123ae378ea",
+      "description": "has matching birthdays without matching birthday pair",
+      "property": "hasMatchingBirthdays",
+      "input": {
+        "birthdays": ["1984-04-05", "2000-09-17", "1966-10-12"]
+      },
+      "expected": false
+    },
+    {
+      "uuid": "89a462a4-4265-4912-9506-fb027913f221",
+      "description": "probability for matching birthday for one person",
+      "scenarios": ["random"],
+      "property": "probabilityForMatchingBirthdays",
+      "input": {
+        "groupSize": 1
+      },
+      "expected": 0.0
+    },
+    {
+      "uuid": "ec31c787-0ebb-4548-970c-5dcb4eadfb5f",
+      "description": "probability for matching birthday for ten people",
+      "scenarios": ["random"],
+      "property": "probabilityForMatchingBirthdays",
+      "input": {
+        "groupSize": 10
+      },
+      "expected": 11.7
```

Rounded to 8 decimal places
```suggestion
      "expected": 11.694818
```